server/cmwell-ws/app/controllers/Application.scala

```diff
@@ -311,20 +311,23 @@ callback=< [URL] >
     }
   }
 
-  private def handleGetForActiveInfoton(req: Request[AnyContent], path: String) = Try {
-    // todo: fix this.. should check that the token is valid...
-    val tokenOpt = authUtils.extractTokenFrom(req)
-    val isRoot = authUtils.isValidatedAs(tokenOpt, "root")
-    val nbg = req.getQueryString("nbg").flatMap(asBoolean).getOrElse(tbg.get)
-
-    val length = req.getQueryString("length").flatMap(asInt).getOrElse(if (req.getQueryString("format").isEmpty) 13 else 0) // default is 0 unless 1st request for the ajax app
-    val offset = req.getQueryString("offset").flatMap(asInt).getOrElse(0)
-    //    infotonOptionToReply(req,ActiveInfotonGenerator.generateInfoton(req.host, path, new DateTime(),length,offset))
-
-    activeInfotonGenerator
-      .generateInfoton(req.host, path, new DateTime(), length, offset, isRoot, nbg)
-      .flatMap(iOpt => infotonOptionToReply(req, iOpt.map(VirtualInfoton.v2i)))
-  }.recover(asyncErrorHandler).get
+  private def handleGetForActiveInfoton(req: Request[AnyContent], path: String) =
+    getQueryString("qp")(req.queryString)
+      .fold(Success(None): Try[Option[RawFieldFilter]])(FieldFilterParser.parseQueryParams(_).map(Some.apply))
+      .map { qpOpt =>
```

Comment:
dangerous. use `flatMap{ qpOpt => Try { ...`

Reply:
I understand. but...
It's the same pattern as `handleSearch`. Is this a mistake also there?

Reply:
Not nessecarily. `authUtils.isValidatedAs` may fail (e.g: unable to retrieve use infoton), but is not called from search...

Though, I won't argue against being more defensive there too.

Reply:
taking it back... it's actually OK. my bad...